src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -29,9 +33,21 @@ public class FunctionsSyncManager : IFunctionsSyncManager, IDisposable
         private const string HubName = "HubName";
         private const string TaskHubName = "taskHubName";
         private const string Connection = "connection";
-        private const string DurableTaskStorageConnectionName = "azureStorageConnectionStringName";
+        private const string DurableTaskV1StorageConnectionName = "azureStorageConnectionStringName";
+        private const string DurableTaskV2StorageOptions = "storageOptions";
+        private const string DurableTaskV2StorageConnectionName = "connectionStringName";
         private const string DurableTask = "durableTask";
 
+        private readonly Regex versionRegex = new Regex(@"Version=(?<majorversion>\d)\.\d\.\d");
```

Comment:
Can't you use [Version.TryParse](https://docs.microsoft.com/en-us/dotnet/api/system.version.tryparse?view=netframework-4.8) for this?

Reply:
The typeName field is of a form like this:

"Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableTaskWebJobsStartup, Microsoft.Azure.WebJobs.Extensions.DurableTask, Version=2.0.0.0, Culture=neutral, PublicKeyToken=014045d636e89289"

The regex will grab the version field from their.

Reply:
The typeName field is of a form like this:

"Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableTaskWebJobsStartup, Microsoft.Azure.WebJobs.Extensions.DurableTask, Version=2.0.0.0, Culture=neutral, PublicKeyToken=014045d636e89289"

The regex is meant to grab the version from that string. I figured since I already had to do the regex to get the version string out, might as well return the major version as part of it.